Amplify/Categories/Auth/AuthCategoryDeviceBehavior.swift

```diff
@@ -7,7 +7,7 @@
 
 import Foundation
 
-public protocol AuthCategoryDeviceBehavior {
+public protocol AuthCategoryDeviceBehavior: class {
```

Comment:
Hmm what was the need for adding this?

Reply:
this was done for API awhile back (https://github.com/aws-amplify/amplify-ios/blob/master/Amplify/Categories/API/ClientBehavior/APICategoryGraphQLBehavior.swift#L9)

i was getting the error for referencing the AuthCategory protocol without the protocol inheriting from `class`. let me pull the error up again

Reply:
```
'weak' must not be applied to non-class-bound 'AuthCategoryBehavior'; consider adding a protocol conformance that has a class bound
```